conjure-http/src/client.rs

```diff
@@ -216,27 +209,82 @@ where
 ///         w.write_all(b"hello world").await.map_err(Error::internal_safe)
 ///     }
 ///
-///     async fn reset(self: Pin<&mut Self>) -> bool
-///     where
-///         W: 'async_trait,
-///     {
+///     async fn reset(self: Pin<&mut Self>) -> bool {
 ///         true
 ///     }
 /// }
 /// ```
-#[async_trait]
 pub trait AsyncWriteBody<W> {
     /// Writes the body out, in its entirety.
     ///
     /// Behavior is unspecified if this method is called twice without a successful call to `reset` in between.
-    async fn write_body(self: Pin<&mut Self>, w: Pin<&mut W>) -> Result<(), Error>;
+    fn write_body(
+        self: Pin<&mut Self>,
+        w: Pin<&mut W>,
+    ) -> impl Future<Output = Result<(), Error>> + Send;
 
     /// Attempts to reset the body so that it can be written out again.
     ///
     /// Returns `true` if successful. Behavior is unspecified if this is not called after a call to `write_body`.
-    async fn reset(self: Pin<&mut Self>) -> bool
+    fn reset(self: Pin<&mut Self>) -> impl Future<Output = bool> + Send;
+}
+
+trait AsyncWriteBodyEraser<W> {
```

Comment:
I know this isn't public, but can we add some docs for what this trait is supposed to be for? It's not very obvious to me

Reply:
Done!